src/api/app/controllers/webui/staging_workflows_controller.rb

```diff
@@ -1,25 +1,44 @@
 class Webui::StagingWorkflowsController < Webui::WebuiController
   layout 'webui2/webui'
 
+  before_action :require_login, except: [:show]
   before_action :set_bootstrap_views
-
   before_action :set_project, only: [:new, :create]
+  after_action :verify_authorized, except: [:show, :new]
 
-  def new; end
+  def new
+    @staging_workflow = StagingWorkflow.new
+    redirect_to staging_workflow_path(@project.staging) if @project.staging
+  end
 
   def create
-    staging_workflow = @project.staging.new
-    if staging_workflow.save
-      flash[:success] = "Staging Workflow for #{@project.name} was successfully created"
-      redirect_to :show
-    else
-      flash[:error] = "Staging Workflow for #{@project.name} couldn't be created"
-      redirect_to :new
+    staging_workflow = @project.build_staging
+
+    authorize staging_workflow
+
+    StagingWorkflow.transaction do
```

Comment:
Can we not use ``:autosave`` on the staging_projects assocation and then get rid of the whole transaction?

```
project = Project.find_or_initialize_by(name: "#{@project.name}:Staging:#{letter}")
staging_workflow.staging_projects << project
staging_workflow.save
```
> Saving of the parent, its associations, and the destruction of marked associations, all happen inside a transaction. This should never leave the database in an inconsistent state.

https://api.rubyonrails.org/classes/ActiveRecord/AutosaveAssociation.html

Reply:
Yes, better! :+1: